tests/test_custom_policy.py

```diff
@@ -28,17 +28,23 @@ def test_custom_offpolicy(model_class, net_arch):
     _ = model_class("MlpPolicy", "Pendulum-v1", policy_kwargs=dict(net_arch=net_arch), learning_starts=100).learn(300)
 
 
-@pytest.mark.parametrize("model_class", [A2C, PPO, SAC, TD3])
+@pytest.mark.parametrize("model_class", [A2C, DQN, PPO, SAC, TD3])
 @pytest.mark.parametrize("optimizer_kwargs", [None, dict(weight_decay=0.0)])
 def test_custom_optimizer(model_class, optimizer_kwargs):
+    # Use different environment for DQN
+    if model_class is DQN:
+        env_id = "CartPole-v1"
+    else:
+        env_id = "Pendulum-v1"
```

Comment:
I'd make this one line for a slightly better reading flow
```suggestion
    env_id = "CartPole-v1" if model_class is DQN else "Pendulum-v1"
```